sdk/identity/azure-identity/inc/azure/identity/export.hpp

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
Why two copies on this macro?  Its part of core and core is part of identity.  Just use it from CORE.  Additionally we could make it "AZ_EXPORT" or "AZURE_EXPORT"

Reply:
No, you don't get it, that's the whole point. We should not reuse, each library has to have its own. So that when you are compiling core, it is seen as `__declspec(dllexport)`. When you are including the very same file from identity, it is seen as `__declspec(dllimport)`.

Reply:
This is a pretty common idiom: https://stackoverflow.com/questions/14980649/macro-for-dllexport-dllimport-switch

Reply:
and it is not a copy, they are named correspondingly to a library - `AZ_CORE_EXPORT`, `AZ_IDENTITY_EXPORT` etc.

Reply:
Ah, the naming is confusing since its an import in some cases.

Reply:
That's also idiomatic. The tings I've seen the most do actually look like this:
```
#ifdef xxx_EXPORTS
#define xxx_EXPORT __declspec(dllexport)
#else
#define xxx_EXPORT __declspec(dllimport)
#endif
```

Reply:
> No, you don't get it, that's the whole point. We should not reuse, each library has to have its own. So that when you are compiling core, it is seen as `__declspec(dllexport)`. When you are including the very same file from identity, it is seen as `__declspec(dllimport)`.

I too was confused by this at first glance before I read your explanation. Anton, consider adding these details and clarifying explanation as a comment.